### PR TITLE
Less global and mutable state

### PR DESCRIPTION
**Pull request by ppalaga**
@gnodet this should be a pure refactoring with no changes of behavior. The main intention is to make the code easier to read and easier to reason about.

### CODE REVIEW

**Comment by gnodet** [comment]
I'm still hesitant to deviate the `DaemonMavenCli` too much from the `MavenCli`...

**Comment by ppalaga** [comment]
> I'm still hesitant to deviate the `DaemonMavenCli` too much from the `MavenCli`...

I agree that keeping `DaemonMavenCli` as close as possible to `MavenCli` would simplify donating the Daemon to ASF. OTOH, `DaemonMavenCli` is the main part of Maven that we customize. Thus, when we start backporting our changes to Maven, `MavenCli` will have to change anyway. In that situation, I believe, it will be better to approch `MavenCli` with a clean design in mind. So if this PR brings a cleaner design to `DaemonMavenCli` any divergence from `MavenCli` should not be considered an obstacle. WDYT?

**Comment by ppalaga** [comment]
I have removed the second commit for now and I have simplified the first one as much as possible. Could you please re-review?

File: daemon/src/main/java/org/apache/maven/cli/DaemonMavenCli.java [hunk 1]
```diff
@@ -808,7 +822,10 @@ private void logSummary(ExceptionSummary summary, Map<String, String> references
 
     private static final String ANSI_RESET = "\u001B\u005Bm";
 
-    private void configure(CliRequest cliRequest)
+    private static void configure(
+            CliRequest cliRequest,
+            EventSpyDispatcher eventSpyDispatcher,
+            Map<String, ConfigurationProcessor> configurationProcessors)
```
Comment (gnodet) [inline daemon/src/main/java/org/apache/maven/cli/DaemonMavenCli.java:828]: Same as above...
Those methods are private and need to be given fields, so it's better to make them non static imho.

File: daemon/src/main/java/org/apache/maven/cli/DaemonMavenCli.java [hunk 2]
```diff
@@ -488,24 +505,18 @@ protected void configure() {
...
-            Set<String> providedArtifacts) {
+    private static List<CoreExtensionEntry> loadCoreExtensions(
+            List<CoreExtension> extensions,
+            ClassRealm containerRealm,
+            Set<String> providedArtifacts,
+            ClassWorld classWorld,
+            Slf4jLoggerManager plexusLoggerManager,
+            Logger slf4jLogger,
+            ILoggerFactory slf4jLoggerFactory,
+            Function<CliRequest, TransferListener> createTransferListener) {
```
Comment (gnodet) [inline daemon/src/main/java/org/apache/maven/cli/DaemonMavenCli.java:519]: What's the benefit of using a static method and passing all parameters ?

Comment (ppalaga) [reply daemon/src/main/java/org/apache/maven/cli/DaemonMavenCli.java:519]: Actually now, that those fields are final, you are right, the method could be non-static without any danger that we inadvertently change the state. Let me change it.

File: daemon/src/main/java/org/apache/maven/cli/DaemonMavenCli.java (hunk 1 above)
Comment (ppalaga) [reply daemon/src/main/java/org/apache/maven/cli/DaemonMavenCli.java:828]: configure is called twice with different params, so it is IMO safer not to allow reading the instance fields.

**Comment by ppalaga** [comment]
2fc3cec: improved what I was able to. Is this good to merge now?

**Comment by gnodet** [comment]
> [2fc3cec](https://github.com/mvndaemon/mvnd/commit/2fc3cec0c20b3676b06c12d355e1f5ab5d0eb5b2): improved what I was able to. Is this good to merge now?

Yes

**Pull request merged**